kitops/modelkit/pydantic_kit.py

```diff
@@ -0,0 +1,195 @@
+from pathlib import Path
+from typing import Any, Optional, Self
+
+from pydantic import BaseModel, Field, model_validator
+
+
+class BasePathModel(BaseModel):
```

Comment:
In all the models I think we need to add `extras=allowed`. So we have some forward compatibility.

Reply:
I'm not a maintainer of the repo, so it's not my call to make, but given the steps taken for validation and existing logic to ensure only permitted keys were in use, it seems to me it would be preferable to add anything new explicitly. If that's not done, then documentation and tests might be lacking, but it also might not be clear to users what is or isn't valid.

I have no issue adding it, but from the outside it doesn't seem like it aligns with the design.